song-server/src/main/java/bio/overture/song/server/controller/AnalysisController.java

```diff
@@ -135,15 +137,16 @@ public void patchUpdateAnalysis(
   @PutMapping(value = "/publish/{id}")
   @SneakyThrows
   @PreAuthorize("@studySecurity.authorize(authentication, #studyId)")
-  public ResponseEntity<String> publishAnalysis(
-      @RequestHeader(value = AUTHORIZATION, required = false) final String accessToken,
+  public ResponseEntity<Object> publishAnalysis(
+      // @RequestHeader(value = AUTHORIZATION, required = false) final String accessToken,
       @PathVariable("studyId") String studyId,
       @PathVariable("id") String id,
       @ApiParam(value = "Ignores files that have an undefined MD5 checksum when publishing")
           @RequestParam(value = "ignoreUndefinedMd5", defaultValue = "false", required = false)
           boolean ignoreUndefinedMd5) {
     val analysis = analysisService.publish(studyId, id, ignoreUndefinedMd5);
-    return ok("AnalysisId " + id + " successfully published");
+    JSONParser parser = new JSONParser();
+    return ok((JSONObject) parser.parse("AnalysisId " + id + " successfully published"));
```

Comment:
can we return the published analysis object instead of a message text?

Reply:
Yes, we can do that too. I will update it accordingly.